zebra/zebra_vty.c

```diff
@@ -422,25 +422,25 @@ static void show_nexthop_detail_helper(struct vty *vty,
 static void zebra_show_ip_route_opaque(struct vty *vty, struct route_entry *re,
 				       struct json_object *json)
 {
+	char obuf[ZAPI_MESSAGE_OPAQUE_LENGTH];
+
 	if (!re->opaque)
 		return;
 
 	switch (re->type) {
 	case ZEBRA_ROUTE_SHARP:
+		strlcpy(obuf, (char *)re->opaque->data, re->opaque->length);
```

Comment:
I think that's got to be "sizeof(obuf)" ?

Reply:
no this was intentionally reduced to the MIN of the two.

Reply:
hmm - you're usually the advocate for good patterns - and it's got to be ok to say "you gotta use the size of the destination buffer in all these sorts of apis" ? I know that there's a trail of breadcrumbs that could be followed back into the zapi code to see whether this construction is ok in this specific case, but ... it could just be un-controversial?

Reply:
I have no idea what you are asking me to do here.  Can you use more words?

Reply:
sorry  - I was suggesting:
`       strlcpy(obuf, (char *)re->opaque->data, sizeof(obuf));`

Reply:
Mark is correct, the last argument to `strlcpy` is the destination buffer size, not the source size. It should be `sizeof(obuf)`, anything else is wrong.

Reply:
fixed